pkg/noderesourcetopology/score.go

```diff
@@ -152,3 +148,29 @@ func containerScopeScore(pod *v1.Pod, zones topologyv1alpha1.ZoneList, scorerFn
 	klog.V(5).InfoS("container scope scoring final node score", "finalScore", finalScore)
 	return finalScore, nil
 }
+
+func (tm *TopologyMatch) scoringHandlerFromTopologyManagerConfig(conf TopologyManagerConfig) scoringFn {
+	if tm.scoreStrategyType == apiconfig.LeastNUMANodes {
+		if conf.Scope == ScopePod {
+			return leastNUMAPodScopeScore
+		}
+		if conf.Scope == ScopeContainer {
+			return leastNUMAContainerScopeScore
+		}
+		return nil // cannot happen
+	}
+	if conf.Policy != PolicySingleNUMANode {
+		return nil
+	}
```

Comment:
I think you can just call `getScoringStrategyFunction` func here and pass it down. By doing it you can remove `scoreStrategyFunc` field from `TopologyMatch` struct, but it would mean that either this function will return additional `error` or we should add some validation in `TopologyMatch` constructor, WDYT?

Reply:
ok, unfortunately I have to take it back, this approach looks nicer and carrying `scoreStrategyFunc` is unfortunate, but it allows us to do the validation in the` TopologyMatch` constructor. Erroring out later in `Score` is too late and can only lead to silent failures - no way to abort, we can only set the score. Let me at least document this rationale with a code comment.

Reply:
Another thing we can try is to implement suggestion from this comment https://github.com/kubernetes-sigs/scheduler-plugins/pull/474#discussion_r1086235462, I think this can vastly simplify things in the future :) or I can just make an issue to add args validation for `TopologyMatch` plugin.

Reply:
👍